--- git-find-md5	2018-05-22 15:19:28.948396878 -0400
+++ git-find-md5	2018-05-22 15:20:05.566483370 -0400
@@ -12,13 +12,15 @@
 fi
 
 # Check if valid git repo
-git status &> /dev/null
+ROOT=$(git status &> /dev/null)
 
 if [[ $? -ne 0 ]]; then
     echo "Not a valid git repo."
     exit 1
 fi
 
+cd "$ROOT"
+
 # git revision for file
 REVS=`git log --pretty=%H -- $FILE`
 
@@ -26,7 +28,7 @@
 for rev in $REVS; do
     git show $rev:$FILE > _file_to_check
     # if you are on a Linux system, change md5 to md5sum
-    if [[ -n `md5 _file_to_check | grep $CHECKSUM` ]]; then
+    if [[ -n `md5sum _file_to_check | grep $CHECKSUM` ]]; then
         echo $rev
     fi
     rm _file_to_check
